packages/upgradeable/src/adapters/EnvironmentAdapter.ts

```diff
@@ -1,23 +1,19 @@
-import { HardhatRuntimeEnvironment } from "hardhat/types";
-import { EnvironmentService, Stage } from "../BaseDeployment.service";
-
+/// <reference path="../types.d.ts"/>
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { EnvironmentService, Stage } from '../BaseDeployment.service';
 const possibleStageValues = Object.values(Stage);
 
 export class EnvironmentAdapter implements EnvironmentService {
+  constructor(readonly hre: HardhatRuntimeEnvironment) {}
 
-    constructor(readonly hre: HardhatRuntimeEnvironment) { }
-
-    /** Returns first tag which matches stage enum or development if nothing is found */
-    async getStage(): Promise<Stage> {
-        const { network } = this.hre;
+  /** Returns first tag which matches stage enum or development if nothing is found */
+  async getStage(): Promise<Stage> {
+    const { network } = this.hre;
 
-        const tags = network.config.tags || [];
+    const tags = network.config.tags || [];
 
-        const stage = tags.find((tag) => 
-            possibleStageValues.includes(tag as Stage)
-        ) as (Stage | undefined);
+    const stage = tags.find((tag) => possibleStageValues.includes(tag as Stage)) as Stage | undefined;
```

Comment:
this not will work properly, result type will be `Stage`

Reply:
Why so?
<img width="626" alt="image" src="https://github.com/eonian-core/farm/assets/10736655/631b588d-eb7f-4c48-8b39-736f7d605f64">